tests/integration_tests/build/test_coverage.py

```diff
@@ -32,6 +32,44 @@
 """Regex for extracting number of total executable lines found by kcov."""
 
 
+@pytest.mark.timeout(120)
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="kcov hangs on aarch64"
+)
+def test_ensure_mod_tests():
+    """Fail if files containing tests don't also have a 'tests' module."""
+    # List all source files containing rust #[test] attribute,
+    # (excluding generated files).
+    # Take the list and check each file contains 'mod tests {', fail if not.
+    cmd = (
+        '/bin/bash '
+        '-c '
+        '"grep '
+        '--files-without-match '
+        '\'mod tests {\' '
+        '\\$(grep '
+        '--files-with-matches '
+        '--recursive '
+        '--exclude-dir=src/*_gen/* '
+        '\'\\#\\[test\\]\' ../src/*/src)" '
+    )
+
+    # the outer grep returns 0 if it finds files without the match, so we
+    # ignore the return codes.
+    result = utils.run_cmd(cmd, no_shell=False, ignore_return_code=True)
+
+    ensure_mod_tests_msg = (
```

Comment:
I would rename this variable to `error_msg`.

Reply:
Done.